### PR TITLE
Add contributing documentation

### PR DESCRIPTION
**Pull request by JakeSCahill**

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+When creating a pull request, please follow these steps to have your contribution considered by the maintainers:
+
+- A pull request should have only one concern (for example one feature or one bug). If a PR addresses more than one concern, it should be split into two or more PRs.
+
+- A pull request can be merged only if it references an open issue
+
+    **Note:** Minor changes such as fixing a typo can but do not need an open issue.
+
+- All code should be well tested
```
Comment (oopsmonk) [inline .github/CONTRIBUTING.md]: The running environment of iota.c could be vary, the functionality can be affected by different hardware architectures, compilers, or operating systems. it's hard to define a `well tested` code or we should have a standard in README or somewhere.
```suggestion
- All code should be covered in unit tests 
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+## Hardware specification
+
+What hardware are you using?
+
+- Operating system:
+- RAM:
+- Number of cores:
+- Device type:
+
```
Comment (oopsmonk) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
## Compiler and build tool specification

What compiler are you using?

What build tool are you using? 
Bazel/Bazelisk/CMake

```

**Review by oopsmonk — COMMENTED** [review]
Many thanks, added some suggestions.

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (JakeSCahill) [reply .github/CONTRIBUTING.md]: Do we mean that code should pass the unit tests (are this in the Git hooks?), or that you should write unit tests for your code?

Comment (oopsmonk) [reply .github/CONTRIBUTING.md]: there are unit tests, like `iota.c/cclient/api/tests`, those will be tested through CI. The contributor might need to add new tests for new functionality. I think it is mentioned in the PR section: https://github.com/iotaledger/iota.c/pull/23/files#diff-8cc7b2b0d78dd2501610391c086a8516R184

**Review by oopsmonk — APPROVED** [review]

**Pull request merged**